packages/app-store/_appRegistry.ts

```diff
@@ -75,13 +76,24 @@ export async function getAppRegistryWithCredentials(userId: number) {
   const apps = [] as (App & {
     credentials: Credential[];
     isDefault?: boolean;
+    dependencyData?: {
+      name?: string;
+      installed?: boolean;
+    };
```

Comment:
When getting all apps for the app store we need to check if the app has a dependency. If so, prevent installing and direct the user to install the dependency